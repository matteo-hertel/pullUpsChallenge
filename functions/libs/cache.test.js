const { processCacheUpdate } = require('./cache');

describe('Cache functions', () => {
  it('should process cache properly', () => {
    const cache = {
      completed: 0,
      rejected: 0,
      weeks: {
        0: {
          completed: 0,
          rejected: 0,
        },
        1: {
          completed: 0,
          rejected: 0,
        },
      },
    };
    const task1 = {
      amount: 10,
      completed: true,
      rejected: false,
      weekNumber: 1,
    };
    const task2 = {
      amount: 10,
      completed: false,
      rejected: true,
      weekNumber: 1,
    };
    const task3 = {
      amount: 10,
      completed: true,
      rejected: false,
      weekNumber: 2,
    };
    const task4 = {
      amount: 10,
      completed: false,
      rejected: true,
      weekNumber: 2,
    };
    const task5 = {
      amount: 100,
      completed: true,
      weekNumber: 8,
    };
    const task6 = {
      amount: 100,
      rejected: true,
      weekNumber: 8,
    };
    expect(
      processCacheUpdate(
        processCacheUpdate(
          processCacheUpdate(
            processCacheUpdate(
              processCacheUpdate(processCacheUpdate(cache, task6), task5),
              task1,
            ),
            task2,
          ),
          task3,
        ),
        task4,
      ),
    ).toMatchSnapshot();
  });
});
