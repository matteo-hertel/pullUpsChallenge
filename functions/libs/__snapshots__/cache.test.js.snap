// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cache functions should process cache properly 1`] = `
Object {
  "completed": 120,
  "rejected": 120,
  "weeks": Object {
    "0": Object {
      "completed": 0,
      "rejected": 0,
    },
    "1": Object {
      "completed": 10,
      "rejected": 10,
    },
    "2": Object {
      "completed": 10,
      "rejected": 10,
    },
    "8": Object {
      "completed": 100,
      "rejected": 100,
    },
  },
}
`;
